### PR TITLE
samples: nrf9160: modem_shell: fix uart disable command

### PR DESCRIPTION
**Pull request by hiltunent**
"uart disable" command was broken. UART driver implementation has been
changed so that PM_DEVICE_STATE_LOW_POWER is not supported anymore.
The "uart disable" command implementation was changed to use
PM_DEVICE_STATE_SUSPENDED instead. Also brought back a deleted
overlay file to enable UART1 in modem shell.

Signed-off-by: Tuomas Hiltunen <tuomas.hiltunen@nordicsemi.no>

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/uart/uart_shell.c [hunk 1]
```diff
@@ -99,13 +99,13 @@ void uart_toggle_power_state(void)
 			/* allow little time for printing the notification */
 			k_sleep(K_MSEC(500));
 
-			/* set uart0 to low power state */
-			err = pm_device_state_set(uart_dev, PM_DEVICE_STATE_LOW_POWER);
```
Comment (jhirsi) [inline samples/nrf9160/modem_shell/src/uart/uart_shell.c:103]: LGTM if this does the required result (=same level pwr consumption than earlier when uarts disabled), but was it clarified that where/why did the PM_DEVICE_STATE_LOW_POWER disappeared from used serial driver?

**Review by jhirsi — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/uart/uart_shell.c (hunk 1 above)
Comment (hiltunent) [reply samples/nrf9160/modem_shell/src/uart/uart_shell.c:103]: Supposedly the driver was "simplified". Actually even before, requesting low power state really caused only a fallthrough to PM_DEVICE_STATE_OFF, see https://github.com/nrfconnect/sdk-zephyr/commit/9e8e88aee4b5d66002b0bf7373ca56b935aecf18

Comment (tokangas) [reply samples/nrf9160/modem_shell/src/uart/uart_shell.c:103]: As background, when I first implemented disabling of the UARTs to modem_shell, the code was changed during the review from setting registers directly to use the Zephyr Device Power Management API. The state suggested in the review was PM_DEVICE_STATE_LOW_POWER and because that worked fine, that one was left in the code. That's the reason why PM_DEVICE_STATE_LOW_POWER was used in the first place.

**Review by tokangas — APPROVED** [review]

File: samples/nrf9160/modem_shell/boards/nrf9160dk_nrf9160_ns.overlay [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
```
Comment (trantanen) [inline samples/nrf9160/modem_shell/boards/nrf9160dk_nrf9160_ns.overlay:1]: Could we put this to a separate commit where we refer to the commit bcf3e8c1f2b70fb791cee7325e1a1d331ee6db94 that removed it accidentally?

**Review by trantanen — APPROVED** [review]

**Comment by hiltunent** [comment]
All concerns have been addressed so I reckon this can be merged now.
